lib/util/utils.js

```diff
@@ -421,15 +421,18 @@ exports.getProvider = function getProvider(path) {
   return result;
 };
 
+/**
 /*
  * Clones a github repository in the given directory.
- * @param {string} github url to be cloned.
+ * @param {string} directory to where to clone the repository.
+ *
+ * @param {string} url of the repository to be cloned.
  *                 Example "https://github.com/Azure/azure-rest-api-specs.git" or
  *                         "git@github.com:Azure/azure-rest-api-specs.git".
- * 
- * @param {string} path where to clone the repository. 
+ *
+ * @param {string} [branch] to be cloned instead of the default branch.
  */
-exports.gitClone = function gitClone(url, directory) {
+exports.gitClone = function gitClone(directory, url, branch) {
```

Comment:
This will break Huiqing's deployments as the method signature has changed. What is the reason to make directory as the first parameter? I understand that you need to add support for branching
How about adding branch as the last parameter to the existing method signature `function gitClone(url, directory, branch) {` ?
This won't break Huiqing's existing scripts.